galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -15,6 +15,2538 @@
+            " FROM     $2                                     " +
+            " WHERE    $1 = $3"
+
+    val SELECT_MATCHING_STRINGS = " SELECT   $1                                     " +
+            " FROM     $2                                     " +
+            " WHERE    {fn SUBSTRING($1, 1, {fn LENGTH(#3)})} = #3    " +
+            " ORDER BY 1"
+
+    if (isNull(block!!.getActiveRecord())) {
+      return
+    }
+    if (list == null) {
+      return
+    }
+    val alreadyProtected: Boolean = getForm().inTransaction()
+    if (this !is VStringField) {
+      var exists: Boolean
+
+      try {
+        while (true) {
+          try {
+            if (!alreadyProtected) {
+              getForm().startProtected(null)
+            }
+            val query = Query(getForm().dBContext.defaultConnection)
+
+            query.addString(list!!.getColumn(0).column)
+            query.addString(evalListTable())
+            query.addString(getSql(block!!.getActiveRecord()))
+            query.open(SELECT_IS_IN_LIST)
+            exists = query.next()
+            query.close()
+            if (!alreadyProtected) {
+              getForm().commitProtected()
+            }
+            break
+          } catch (e: SQLException) {
+            if (!alreadyProtected) {
+              getForm().abortProtected(e)
+            } else {
+              throw e
+            }
+          } catch (error: Error) {
+            if (!alreadyProtected) {
+              getForm().abortProtected(error)
+            } else {
+              throw error
+            }
+          } catch (rte: RuntimeException) {
+            if (!alreadyProtected) {
+              getForm().abortProtected(rte)
+            } else {
+              throw rte
+            }
+          }
+        }
+      } catch (e: Throwable) {
+        throw VExecFailedException(e)
+      }
+      if (!exists) {
+        throw VFieldException(this, MessageCode.getMessage("VIS-00001"))
+      }
+      return
+    } else {
+      var query: Query
+      var count = 0
+      var result: String? = null
+      val fldbuf: String = getSql(block!!.getActiveRecord())
+
+      if (fldbuf.indexOf('*') > 0) {
+        return
+      }
+      try {
+        while (true) {
+          try {
+            if (!alreadyProtected) {
+              getForm().startProtected(null)
+            }
+            query = Query(getForm().dBContext.defaultConnection)
+            query.addString(list!!.getColumn(0).column)
+            query.addString(evalListTable())
+            query.addString(getString(block!!.getActiveRecord()))
+            query.open(SELECT_MATCHING_STRINGS)
+            if (!query.next()) {
+              count = 0
+            } else {
+              count = 1
+              result = query.getString(1)
+              if (query.next()) {
+                count = 2
+              }
+            }
+            query.close()
```

Comment:
use  Query(getForm().dBContext.defaultConnection).let {.. }

Reply:
query used in if (!query.next()) {